upgrade/scripts/upgrade/util/postgres-reinit.sh

```diff
@@ -0,0 +1,174 @@
+      sleep ${wait}
+      count=$((count + 1))
+      continue
+    fi
+  done
+
+  if [ -z ${pg_pod} ]; then
+    echo "ERROR: Unable to recover ${pg_cluster} in ${pg_ns} automatically. You may have to re-install the chart."
+    exit 1
+  fi
+
+  count=1
+  while [ ${count} -le ${total} ]; do
+    # Make sure leader exists for the recovery
+    if kubectl exec -i ${pg_pod} -n ${pg_ns} -- patronictl list | grep "${pg_cluster}-" | grep Leader | grep running; then
+      leader=$(kubectl exec -i ${pg_pod} -n ${pg_ns} -- patronictl list | grep "${pg_cluster}-" | grep Leader | awk '{print $2}')
+      echo "INFO: found the leader pod ${leader}"
+      # reinit instances not in "running" state
+      reinit_count=1
+      reinit_max=5
+      while kubectl exec -i ${leader} -n ${pg_ns} -- patronictl list | grep "${pg_cluster}-" | grep -v -m 1 "running"; do
+        # get the first instance not in running state
+        failed_instance=$(kubectl exec -i ${leader} -n ${pg_ns} -- patronictl list | grep "${pg_cluster}-" | grep -v -m 1 "running" | awk '{print $2}')
+        echo "Reinit ${failed_instance} not in running state: attempt ${reinit_count}/${reinit_max}..."
+        echo "y" | kubectl exec -i ${leader} -n ${pg_ns} -- patronictl reinit ${pg_cluster} ${failed_instance} 2> /dev/null || true
+        # sleep for some time as reinit takes a while to settle
+        sleep ${wait}
+        reinit_count=$((reinit_count + 1))
+        if [ ${reinit_count} -gt ${reinit_max} ]; then
+          echo "Reinit count reached ${reinit_max}. Breaking loop."
+          break
+        fi
+      done
+
+      # Reinit instances with replication lag
+      reinit_count=1
+      reinit_max=3
+      while [ "$(kubectl exec -i ${leader} -n ${pg_ns} -- patronictl list | grep Replica | awk '{print $12}' | sort -u)" != "0" ]; do
+        for line in $(kubectl exec -i ${leader} -n ${pg_ns} -- patronictl list | grep Replica); do
+          if [ "$(echo "$line" | awk '{print $12}')" != "0" ]; then
+            lagged_instance=$(echo $line | awk '{print $2}')
+            echo "Reinit ${lagged_instance} with replication lag: attempt ${reinit_count}/${reinit_max}..."
+            echo "y" | kubectl exec -i ${leader} -n ${pg_ns} -- patronictl reinit ${pg_cluster} ${lagged_instance} 2> /dev/null || true
+            # sleep for some time as reinit takes a while to settle
+            sleep ${wait}
+          fi
+        done
+        reinit_count=$((reinit_count + 1))
+        if [ ${reinit_count} -gt ${reinit_max} ]; then
+          echo "Reinit count reached ${reinit_max}. Breaking loop."
+          break
+        fi
+      done
+    else
+      echo "WARNING: No running leader is found for ${pg_cluster} in ${pg_ns}. Rollout restart it and sleep for ${wait} seconds, attempt ${count}/${total})..."
+      kubectl rollout restart sts ${pg_cluster} -n ${pg_ns}
+      sleep ${wait}
+      count=$((count + 1))
+      continue
+    fi
+    break
+  done
+  if [ -z ${leader} ]; then
```

Comment:
Need to add double quotes or it'll be a syntax error when it is empty
```suggestion
  if [ -z "${leader}" ]; then
```